File: 025주차/k진수에서 소수 개수 구하기/woosteelz.py
import math


def is_prime_number(x):
    if x == 1:
        return False
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    return True


def change_k(num, k):
    answer = ''
    while num:
        num, mod = divmod(num, k)
        answer += str(mod)
    return answer[::-1]


def solution(n, k):
    answer = 0
    n = change_k(n, k)

    prime_list = n.split('0')

    for prime in prime_list:
        if prime and is_prime_number(int(prime)):
            answer += 1

    return answer

File: 025주차/k진수에서 소수 개수 구하기/woosteelz.js
const isPrimeNum = (num) => {
    if (num === 1) {
        return false;
    }
    for (let i = 2; i <= Math.round(Math.sqrt(num)); i++) {
        if (num % i === 0) {
            return false;
        }
    }
    return true;
}

const solution = (n, k) => {
    let ans = 0;

    const num = n.toString(k); // 10진수를 k진수로 변환 (문자열 리턴)
    // const num = parseInt(n, k); // k진수인 n을 10진수로 변환 (정수형 리턴)

    const PrimeList = num.split("0");

    PrimeList.map(function (num) {
        if (num && isPrimeNum(Number(num))) ans++;
    })
    return ans;
}

const n = 100
console.log(isPrimeNum(2))